apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/sampling/SamplingRateWatcher.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.sampling;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.conf.dynamic.AgentConfigChangeWatcher;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+
+public class SamplingRateWatcher extends AgentConfigChangeWatcher {
+
+    private AtomicInteger samplingRate;
+    private final SamplingService samplingService;
+
+    private static final ILog LOGGER = LogManager.getLogger(SamplingRateWatcher.class);
+
+    public SamplingRateWatcher(final String propertyKey, SamplingService samplingService) {
+        super(propertyKey);
+        this.samplingRate = new AtomicInteger(getDefaultValue());
+        this.samplingService = samplingService;
+    }
+
+    private void activeSetting(String config) {
+        if (LOGGER.isDebugEnable()) {
+            LOGGER.debug("Updating using new static config: {}", config);
+        }
+        try {
+            this.samplingRate = new AtomicInteger(Integer.parseInt(config));
```

Comment:
You should not recreate an `AtomicInteger`, which is a complete misuse of `AtomicInteger`.

```suggestion
            this.samplingRate.set(Integer.parseInt(config));
```

if you want it to be safer, use `compareAndSet` to make sure it’s not set by other thread (it’s rare here but in case )

Reply:
I think set is enough.